lib/oidcstrategy.js

```diff
@@ -69,18 +68,22 @@ const NONCE_MAX_AMOUNT = 10;
 const NONCE_LIFE_TIME = 3600; // second
 
 function makeProfileObject(src, raw) {
+  var emails = [];
+  if (src.upn)
+    emails = [src.upn];
+  else if (src.emails)
+    emails = src.emails;
```

Comment:
no. For B2C we get `emails` but not `upn`; for non B2C we get `upn` but no `emails`